persistence-cassandra-4.0/src/main/java/io/stargate/db/cassandra/impl/RowDecoratorImpl.java

```diff
@@ -43,17 +54,98 @@ private RowDecoratorImpl(TableName tableName, TableMetadata tableMetadata) {
+  }
+
   @Override
-  protected ComparableKey<?> decoratePrimaryKey(Object... pkValues) {
-    Clustering key = metadata.partitionKeyAsClusteringComparator().make(pkValues);
-    DecoratedKey decoratedKey = metadata.partitioner.decorateKey(key.serializeAsPartitionKey());
+  protected ComparableKey<?> decoratePrimaryKey(Object... rawKeyValues) {
+    DecoratedKey decoratedKey = decoratedKeyFromRawKeyValues(rawKeyValues);
     return new ComparableKey<>(PartitionPosition.class, decoratedKey);
   }
 
+  /**
+   * {@inheritDoc}
+   *
+   * <p><b>IMPORTANT:</b> we only support getting the bytes for the Murmur3Partitioner,
+   * ByteOrderedPartitioner & OrderPreservingPartitioner. Any other partitioner will only depend on
+   * the key comparison, which is not going to work for the correct OR queries in the Docs API V2.
+   */
   @Override
   public ByteBuffer getComparableBytes(Row row) {
-    // TODO replace this with the relevant row's byte-comparable value when
-    // https://github.com/apache/cassandra/pull/1294 is ready
-    return ByteBuffer.wrap(new byte[] {});
+    DecoratedKey decoratedKey = decoratedKeyFromRawKeyValues(primaryKeyValues(row));
+
+    // based on
+    // https://github.com/apache/cassandra/blob/trunk/src/java/org/apache/cassandra/db/DecoratedKey.java#L109-L119
+
+    ByteComparable.Version version = ByteComparable.Version.OSS42;
+    ByteSource keyComparableBytes = ByteSource.of(decoratedKey.getKey(), version);
+    ByteSource tokenComparableBytes = getTokenComparableBytes(decoratedKey.getToken(), version);
+
+    // combine
+    ByteSource bytes =
+        ByteSource.withTerminatorMaybeLegacy(
+            version, ByteSource.END_OF_STREAM, tokenComparableBytes, keyComparableBytes);
+    return ByteBuffer.wrap(readBytes(bytes, 64));
+  }
+
+  // resolves the comparable bytes for three token types
+  private static ByteSource getTokenComparableBytes(Token token, ByteComparable.Version version) {
+    // https://github.com/apache/cassandra/blob/trunk/src/java/org/apache/cassandra/dht/ByteOrderedPartitioner.java#L107-L111
+    if (token instanceof ByteOrderedPartitioner.BytesToken) {
+      byte[] tokenValue = (byte[]) token.getTokenValue();
+      return ByteSource.of(tokenValue, version);
+    }
+
+    // https://github.com/apache/cassandra/blob/trunk/src/java/org/apache/cassandra/dht/Murmur3Partitioner.java#L182-L186
+    if (token instanceof Murmur3Partitioner.LongToken) {
+      long tokenValue = (long) token.getTokenValue();
+      return ByteSource.of(tokenValue);
+    }
+
+    // https://github.com/apache/cassandra/blob/trunk/src/java/org/apache/cassandra/dht/OrderPreservingPartitioner.java#L206-L210
+    if (token instanceof OrderPreservingPartitioner.StringToken) {
+      String tokenValue = (String) token.getTokenValue();
+      return ByteSource.of(tokenValue, version);
+    }
+
+    // we only support the above 3
+    // for other partitioner, log warn and wrap empty bytes
+    LOG.warn(
```

Comment:
Is this only warning and not error wrt logging? Do we expect that this would actually happen?

Reply:
And with all that, comparable bytes are only needed for `$or` searches.. Thus, I just actually created https://github.com/stargate/stargate/issues/1960